OpenSim/Common/TableSource.h

```diff
@@ -95,21 +121,52 @@ class TableSource_ : public Component {
         return _table;
     }
 
-    /** Replace the existing TimeSeriesTable_ this TableSource_ currently 
-    holds.                                                                    
+    /** Replace the existing TimeSeriesTable_ that this TableSource_ currently 
+    holds. This operation is not allowed if TableSource_ already contains a
+    value for the property `filename`. In such cases, update the property
+    'filename' using `upd_filename()` to empty string and call this function.
 
-    \throws KeyNotFound If table provided does not have column-labels.        */
+    \throws InvalidCall If property `filename` is set. This call is not allowed
+                        if `filename` property is set.                        
+    \throws KeyNotFound If table provided does not have column labels.        */
     void setTable(const Table& table) {
+        OPENSIM_THROW_IF(!get_filename().empty(),
+                         InvalidCall,
+                         "Property 'filename' is set. Cannot set Table now.");
```

Comment:
I like this check here. I think the user may be doing something wrong if they set the table programmatically and in XML. It may be an accident; which table did they really want to use? If they didn't want to use the one specified in XML, then the filename should be empty. One option is to have a second argument to this method, like `bool overrideIfFilenameGiven`.